src/commonMain/kotlin/org/mongodb/kbson/serialization/Serializers.kt

```diff
@@ -394,9 +397,6 @@ internal object BsonValueSerializer : KSerializer<BsonValue>, BsonSerializer {
                 jsonElement.doubleOrNull?.let {
                     return BsonDouble(it)
                 }
-                jsonElement.contentOrNull?.let {
```

Comment:
Should we still keep this as a fallback? Otherwise, if the JsonPrimitive introduces another type at any point, it will start returning `null` rather than the value?